hal_st/stm32fxxx/GpioStm.hpp

```diff
@@ -249,6 +255,7 @@ namespace hal
             Port port;
             uint8_t pin;
             uint8_t channel;
+            AdcChannelType channelType{ AdcChannelType::P };
```

Comment:
Would it be possible to use feature testing of some sorts? I would hate to having to spend additional overhead for something only used on one line of chips.

Reply:
Removed `AdcChannelType` idea